src/Storages/CompressionCodecSelector.h

```diff
@@ -92,7 +94,14 @@ class CompressionCodecSelector
 
         for (const auto & element : elements)
             if (element.check(part_size, part_size_ratio))
+            {
                 res = factory.get(element.family_name, element.level);
+                if(res->isDeflateQplCompression() && !element.enable_deflate_qpl_codec)
```

Comment:
Please move this check into method `element.check()` (which already performs checks for part_size and part_size_ratio)